### PR TITLE
Splash Window Modifications...

### PR DESCRIPTION
**Pull request by mamarilmanson**
- Made Settings and New buttons show a "modified" pop_window...
- Have pointing hand cursor when hovering over "Settings" and "New" buttons
- Some reformatting of variable declarations and putting section header comments for quick findings
- Connect in ready as seperate line from anon functions (I don't know if you like it but I find it as organized)

Question: Where can I find the 3d model you were using for tracking? I can't find it... Thank you...

:)

### CODE REVIEW

**Comment by mamarilmanson** [comment]
RunnerItem.tscn is for preparation of getting runners from dummy or actual saved items... :)

File: screens/home/home.gd [hunk 1]
```diff
@@ -1,8 +1,18 @@
 extends CanvasLayer
 
+
+#-----------------------------------------------------------------------------#
```
Comment (you-win) [inline screens/home/home.gd:4]: Can you remove the extra formatting comments? The original `builtin`, `private`, `public` comments were for denoting access like in languages like Java and C++.

File: screens/home/home.gd [hunk 2]
```diff
@@ -12,28 +22,18 @@ const SortDirection := {
...
-@onready
-var _duck: TextureRect = %Duck
-@onready
-var _logo: TextureRect = %Logo
-@onready
-var _sub_logo: TextureRect = %SubLogo
-
-@onready
-var _parallax_elements: Array[TextureRect] = [
+@onready var _viewport: Viewport = get_viewport()
```
Comment (you-win) [inline screens/home/home.gd:27]: Can you revert these changes? Annotations above variables is the code-style used throughout the project.

File: screens/home/home.gd [hunk 3]
```diff
@@ -342,6 +340,16 @@ func _ready() -> void:
 #		sort_runners_popup.index_pressed.emit(0)
 	)
 
+
+func ShowPopups(title: String, scene: PackedScene):
```
Comment (you-win) [inline screens/home/home.gd:344]: The func is in the wrong section, it should likely be a private function.

The name should also try to follow Godot's style, which is `snake_case` with a a leading `_` to denote that it's a private func. It should also have a return type i.e. `func _show_popups(title: String, scene: PackedScene) -> void:`.

File: screens/popup_window_new.gd [hunk 4]
```diff
@@ -0,0 +1,41 @@
+class_name PopupWindowNew
```
Comment (you-win) [inline screens/popup_window_new.gd:1]: Why is this file needed? There is already a `PopupWindow` class at `screens/popup_window.gd`

File: screens/New/New.tscn [hunk 5]
```diff
@@ -0,0 +1,15 @@
+[gd_scene format=3 uid="uid://cq0g8yat1gtw6"]
```
Comment (you-win) [inline screens/New/New.tscn:1]: Please try to follow [Godot's naming convention](https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#file-names) of `snake_case` for file names.

Also, the project uses `kebab-case` for folder names, not `PascalCase`.

Is a new folder necessary for this file? If it's only used on the `home` screen, it should be placed under `screens/home`.

File: screens/Settings/Settings.tscn [hunk 6]
```diff
@@ -0,0 +1,15 @@
+[gd_scene format=3 uid="uid://cxumlykbx3bwe"]
```
Comment (you-win) [inline screens/Settings/Settings.tscn:1]: `snake_case` file name. `kebab-case` folder names.

Settings can probably be left in its own folder, since it should be accessible from many parts of the application.

File: screens/home/home.gd [hunk 7]
```diff
@@ -44,24 +44,42 @@ var _parallax_elements: Array[TextureRect] = [
 ## TextureRect -> Vector2
 var _parallax_initial_positions := {}
 
-@onready
-var _runner_container: PanelContainer = %RunnerContainer
-@onready
-var _runners: VBoxContainer = %Runners
+
+#-----------------------------------------------------------------------------#
```
Comment (you-win) [inline screens/home/home.gd:48]: Revert these changes.

File: screens/home/home.gd [hunk 8]
```diff
@@ -90,27 +108,8 @@ func _ready() -> void:
...
-#		_settings_popup.add_child(settings)
-#
-#		add_child(_settings_popup)
-#		_settings_popup.popup_centered_ratio(0.5)
-#
-#		var settings_hide := handle_popup_hide.bind(_settings_popup)
-#		_settings_popup.visibility_changed.connect(settings_hide)
-#		_settings_popup.close_requested.connect(settings_hide)
-#	)
-	var sort_direction: Button = %SortDirection
```
Comment (you-win) [inline screens/home/home.gd:112]: Generally, if a node does not need to be accessed many times or if a node only needs to be setup in `_ready`, I try to keep them local.

This is to help organize the code like:
- class variable: used many times throughout the class
- local variable: used locally

Can this change be reverted.

File: screens/home/home.gd [hunk 9]
```diff
@@ -182,8 +181,7 @@ func _ready() -> void:
 		for data in (func() -> Array:
 			var r := []
 
-			const USER_DATA_DIR := "user://"
```
Comment (you-win) [inline screens/home/home.gd:185]: This is for testing only as mentioned in the TODO comment above, can this be reverted so it's easier to track

File: components/RunnerItem.tscn [hunk 10]
```diff
@@ -0,0 +1,4 @@
+[gd_scene format=3 uid="uid://dtropu4wnggwb"]
```
Comment (you-win) [inline components/RunnerItem.tscn:1]: Is a new `components` folder necessary? There is already a `RunnerItem` file at `screens/home/runner_item.tscn` and `screens/home/runner_item.gd`

It's better to keep code next to the screens where they are used, unless that code is going to be used elsewhere in the project.

**Review by you-win — CHANGES_REQUESTED** [review]

**Comment by you-win** [comment]
Thanks for the PR!

> Question: Where can I find the 3d model you were using for tracking? I can't find it... Thank you...

The test model is from the [godot-vrm](https://github.com/V-Sekai/godot-vrm) repo. It's actually a [vrm reference model](https://github.com/vrm-c/vrm-specification) but it appears they removed it from that repo.

I've left some comments directly on the files.

In the future, you will want to work off of a git branch rather than `master`, since that will make syncing your changes with this repo difficult for you if `vpuppr:master` changes while you have commits on `mamarilmanson:master`.

File: screens/popup_window_new.gd (hunk 4 above)
Comment (mamarilmanson) [reply screens/popup_window_new.gd:1]: It is not needed... it was only temporary as I was not able to trace where context was coming from...

File: components/RunnerItem.tscn (hunk 10 above)
Comment (mamarilmanson) [reply components/RunnerItem.tscn:1]: I see... it was a habbit of mine from using react and vue where small components were within a "components"... will revert/remove changes...

**Comment by mamarilmanson** [comment]
> 

Still new to git stuff... so I'm lost in every step of the way... :)

File: screens/Settings/Settings.tscn (hunk 6 above)
Comment (you-win) [reply screens/Settings/Settings.tscn:1]: Can this be fixed?

File: screens/home/home.gd (hunk 1 above)
Comment (you-win) [reply screens/home/home.gd:4]: Can this be fixed?

File: screens/home/home.gd [hunk 11]
```diff
@@ -1,8 +1,18 @@
...
+#-----------------------------------------------------------------------------#
+# Scene Loads
+#-----------------------------------------------------------------------------#
 const RunnerItem: PackedScene = preload("res://screens/home/runner_item.tscn")
 const NewRunner: PackedScene = preload("res://screens/home/new_runner.tscn")
+const SettingsScene: PackedScene = preload("res://screens/settings/settings.tscn")
+const CreateNewsScene: PackedScene = preload("res://screens/home/new.tscn")
 
+
+#-----------------------------------------------------------------------------#
```
Comment (you-win) [inline screens/home/home.gd:13]: This section comment should be reverted

File: screens/home/home.gd [hunk 12]
```diff
@@ -31,37 +46,38 @@ var _duck: TextureRect = %Duck
...
 	_logo,
 	_sub_logo
 ]
-## Initial positions for all parallax elements.
-## [br][br]
-## TextureRect -> Vector2
-var _parallax_initial_positions := {}
-
+@onready
+var root_node: CanvasLayer = $"."
```
Comment (you-win) [inline screens/home/home.gd:56]: This isn't needed since `root_node == self`.

File: screens/home/home.gd (hunk 8 above)
Comment (you-win) [reply screens/home/home.gd:112]: Can this be fixed

File: screens/home/home.gd [hunk 13]
```diff
@@ -342,6 +339,16 @@ func _ready() -> void:
 #		sort_runners_popup.index_pressed.emit(0)
 	)
 
+
+func _show_popups(title: String, scene: PackedScene):
```
Comment (you-win) [inline screens/home/home.gd:343]: This should be moved to the appropriate section, which is the private methods section. It also needs an explicit return value.

File: screens/popup_window_new.gd (hunk 4 above)
Comment (you-win) [reply screens/popup_window_new.gd:1]: This file should be removed then

File: screens/home/new.tscn [hunk 14]
```diff
@@ -0,0 +1,15 @@
+[gd_scene format=3 uid="uid://cq0g8yat1gtw6"]
```
Comment (you-win) [inline screens/home/new.tscn:1]: At this point, this file should be removed. I already have something locally that can serve as the scene.

File: screens/home/home.gd [hunk 15]
```diff
@@ -31,37 +46,38 @@ var _duck: TextureRect = %Duck
...
-## TextureRect -> Vector2
-var _parallax_initial_positions := {}
-
+@onready
+var root_node: CanvasLayer = $"."
 @onready
 var _runner_container: PanelContainer = %RunnerContainer
 @onready
 var _runners: VBoxContainer = %Runners
+@onready
```
Comment (you-win) [inline screens/home/home.gd:61]: These variables should be private.

**Review by you-win — CHANGES_REQUESTED** [review]
